web/docs/general/typescript.md

```diff
@@ -0,0 +1,159 @@
+---
+title: TypeScript Support
+---
+
+import TypescriptServerNote from '../_TypescriptServerNote.md'
+
+# TypeScript support
+
+TypeScript is a programming language that adds static type analysis to JavaScript.
+It is a superset of JavaScript, which means all JavaScript code is valid TypeScript code.
+It also compiles to JavaScript before running.
+
+TypeScript's type system helps catch errors at build time (this reduces runtime errors), and provides type-based auto-completion in IDEs.
+
+Each Wasp feature includes TypeScript documentation.
+
+If you're starting a new project and want to use TypeScript, you don't need to do anything special.
+Just follow the feature docs you are interested in, and they will tell you everything you need to know.
+We recommend you start by going through [the tutorial](../tutorial/01-create.md).
+
+To migrate an existing Wasp project from JavaScript to TypeScript, follow this guide.
+
+## Migrating your project to TypeScript
+
+Wasp supports TypeScript out of the box.
```

Comment:
This sentence feels loose, floating alone up here. Maybe remove it or join it with the text right below it.

Reply:
I tried moving it down, but it looked weirder:
![image](https://github.com/user-attachments/assets/e2c4bb32-ac68-4c52-b903-4f75a341784c)

So I changed the text to this:
![image](https://github.com/user-attachments/assets/94ec45e5-f203-464e-b5b3-8ddb8574c0de)